### PR TITLE
Add a test server

### PR DESCRIPTION
**Pull request by arv**
This exposes the Repm module Dispatch function as a web server. It is
designed to be used with Dart/Flutter unit tests.

Towards https://github.com/rocicorp/replicache/issues/46

### CODE REVIEW

**Comment by arv** [comment]
TBR

File: cmd/test_server/main.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+	"gopkg.in/alecthomas/kingpin.v2"
+	"roci.dev/replicache-client/repm"
+)
+
+func main() {
+	impl(os.Args[1:], os.Stdin, os.Stdout, os.Stderr, os.Exit)
+}
+
+func impl(args []string, in io.Reader, out, errs io.Writer, exit func(int)) {
```
Comment (aboodman) [inline cmd/test_server/main.go:19]: The one thing I didn't consider with this approach is that if we ever want push bindings, we'll need to upgrade this to web sockets. Do you see any problem with that?

File: cmd/test_server/main.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+	http.ListenAndServe(fmt.Sprintf(":%d", *port), s)
+}
+
+type testServer struct {
+	dispatch func(dbName, rpc string, data []byte) (ret []byte, err error)
+}
+
+func (s testServer) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	fmt.Println(req.URL.String())
+	if req.URL.Path == "/statusz" {
```
Comment (aboodman) [inline cmd/test_server/main.go:49]: [help] Why do we need this?

**Comment by aboodman** [comment]
See https://news.ycombinator.com/item?id=23027988 for meaning of tags.

File: cmd/test_server/main.go (hunk 2 above)
Comment (arv) [reply cmd/test_server/main.go:49]: The test pings this before running any test functions to make sure the server is up and running.

File: cmd/test_server/main.go (hunk 1 above)
Comment (arv) [reply cmd/test_server/main.go:19]: Not sure. Will have to think about it further.

**Pull request merged**